unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Scene/Messaging/MessagingControllersManager.cs

```diff
@@ -72,82 +70,87 @@ public void Initialize()
 
         public void PopulateBusesToBeProcessed()
         {
-            IWorldState worldState = Environment.i.world.state;
-            string currentSceneId = worldState.currentSceneId;
-            List<IParcelScene> scenesSortedByDistance = worldState.scenesSortedByDistance;
-
-            int count = scenesSortedByDistance.Count; // we need to retrieve list count everytime because it
-            // may change after a yield return
+            lock (sortedControllers)
+            lock (busesToProcess)
```

Comment:
Does it makes sense to move the `MessagingControllersManager` and `MessagingBus` logic to threads? We found bottlenecks here?